### PR TITLE
Add RNN to TMVA

### PR DESCRIPTION
**Pull request by lmoneta**
This PR include the RNN developed by Saurav Shektar, GSOC 2017

### CODE REVIEW

File: cmake/modules/RootBuildOptions.cmake [hunk 1]
```diff
@@ -167,6 +167,8 @@ ROOT_BUILD_OPTION(table OFF "Build libTable contrib library")
 ROOT_BUILD_OPTION(tcmalloc OFF "Using the tcmalloc allocator")
 ROOT_BUILD_OPTION(thread ON "Using thread library (cannot be disabled)")
 ROOT_BUILD_OPTION(tmva ON "Build TMVA multi variate analysis library")
+ROOT_BUILD_OPTION(tmva-cpu ON "Build TMVA with CPU support for deep learning. Requires BLAS")
+ROOT_BUILD_OPTION(tmva-gpu ON "Build TMVA with GPU support for deep learning. Requries CUDA")
```
Comment (amadio) [inline cmake/modules/RootBuildOptions.cmake]: Do we really need the extra CMake options? We can probably reuse the `cuda` option, and add only `blas` to the list. What do you think?

File: cmake/modules/RootConfiguration.cmake [hunk 2]
```diff
@@ -565,6 +565,17 @@ if(qt5web)
 else()
   set(hasqt5webengine undef)
 endif()
+if (tmva AND imt AND BLAS_FOUND)
+  set(hastmvacpu define)
+else()
+  set(hastmvacpu undef)
+endif()  
+if (tmva AND CUDA_FOUND)
```
Comment (Axel-Naumann) [inline cmake/modules/RootConfiguration.cmake]: `if (tmva-gpu)`?

File: tmva/tmva/CMakeLists.txt [hunk 3]
```diff
@@ -73,14 +73,14 @@ endif()
 #---Handle BLAS dependent code. -----------------
 if(BLAS_FOUND AND imt)
     message(STATUS "Using TMVA-DNN with BLAS installation") 
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DDNNCPU")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
```
Comment (Axel-Naumann) [inline tmva/tmva/CMakeLists.txt]: Remove?

File: tmva/tmva/inc/TMVA/DNN/DeepNet.h [hunk 4]
```diff
@@ -58,6 +58,7 @@ using namespace TMVA::DNN::DAE;
 #include <cmath>
 
 using namespace TMVA::DNN::CNN;
+using namespace TMVA::DNN::RNN;
```
Comment (Axel-Naumann) [inline tmva/tmva/inc/TMVA/DNN/DeepNet.h]: Don't put `using namespace` at global scope in headers, please!

**Comment by lmoneta** [comment]
@phsft-bot build with flags -Dimt=On

**Pull request merged**